tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -290,51 +290,12 @@ public static EntryRecordMode GetRecordMode(HttpRequest request)
             return mode;
         }
 
-        private byte[] CompressBody(byte[] incomingBody, SortedDictionary<string, string[]> headers)
-        {
-            if (headers.TryGetValue("Content-Encoding", out var values))
-            {
-                if (values.Contains("gzip"))
-                {
-                    using (var uncompressedStream = new MemoryStream(incomingBody))
-                    using (var resultStream = new MemoryStream())
-                    {
-                        using (var compressedStream = new GZipStream(resultStream, CompressionMode.Compress))
-                        {
-                            uncompressedStream.CopyTo(compressedStream);
-                        }
-                        return resultStream.ToArray();
-                    }
-                }
-            }
-
-            return incomingBody;
-        }
-
-        private byte[] DecompressBody(MemoryStream incomingBody, HttpContentHeaders headers)
-        {
-            if (headers.TryGetValues("Content-Encoding", out var values))
-            {
-                if (values.Contains("gzip"))
-                {
-                    using (var uncompressedStream = new GZipStream(incomingBody, CompressionMode.Decompress))
-                    using (var resultStream = new MemoryStream())
-                    {
-                        uncompressedStream.CopyTo(resultStream);
-                        return resultStream.ToArray();
-                    }
-                }
-            }
-
-            return incomingBody.ToArray();
-        }
-
-        public HttpRequestMessage CreateUpstreamRequest(HttpRequest incomingRequest, byte[] incomingBody)
+        public HttpRequestMessage CreateUpstreamRequest(HttpRequest incomingRequest)
         {
             var upstreamRequest = new HttpRequestMessage();
             upstreamRequest.RequestUri = GetRequestUri(incomingRequest);
             upstreamRequest.Method = new HttpMethod(incomingRequest.Method);
-            upstreamRequest.Content = new ReadOnlyMemoryContent(incomingBody);
+            upstreamRequest.Content = new StreamContent(incomingRequest.Body);
```

Comment:
This change was made because we now use the body directly from the incoming request, rather than the body from the RecordEntry, as the RecordEntry body would be decompressed.

Reply:
@JoshLove-msft there is something going wrong here. I hit issues with the JS test-utils. This is the raw request that is generated:

```
POST http://127.0.0.1:8080/sample_response/abcdef HTTP/1.1
Host: 127.0.0.1:8080
Connection: keep-alive
User-Agent: core-rest-pipeline/1.9.3 Node/v16.11.1 OS/(x64-Windows_NT-10.0.19044)
Accept-Encoding: gzip,deflate
x-ms-client-request-id: 9cfe9d23-0d2a-4438-8905-f568295da7ae
traceparent: 00-7ad45034777edc7bc1e007d10955a25b-521a4c601c0706e7-00
Content-Type: text/plain
Content-Length: 6
```

And we _expect_ the body to contain `abcdef` which is why the Content-Length is 6. For some reason that's not what we're sending.

Reply:
Aha. I don't think we reset body position, and DecompressBody moves it forward. So when we pass the stream to StreamContent is doesn't get reset.

Reply:
Unfortunately, I think we're going to have to use the previous methodology. I don't really want to [enable buffering](https://devblogs.microsoft.com/dotnet/re-reading-asp-net-core-request-bodies-with-enablebuffering/) for perf reasons.

Reply:
Ah I see. I will push out a fix, and hopefully a test.

Reply:
Yeah this is exactly what it was. Unfortunately, during `record`, we effectively _never_ send a request body. We still have the headers specifying content-length, but because we have exhausted the stream to read it into the entry, when we pass the stream over to the response body, it just writes 0 bytes.

I have pushed an update, your refactor of the GZip utility worked great, even if we do a tiny bit of additional work during `recording`.

Reply:
Thanks, I can update the MockHttpHandler so we have some coverage of this scenario.